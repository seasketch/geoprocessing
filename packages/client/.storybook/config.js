import { configure } from "@storybook/react";
import { addDecorator } from "@storybook/react";
import { withKnobs } from "@storybook/addon-knobs";

// automatically import all files ending in *.stories.tsx
const req = require.context("../src/components", true, /\.stories\..+$/);

function loadStories() {
  req.keys().forEach(req);
}

configure(loadStories, module);
