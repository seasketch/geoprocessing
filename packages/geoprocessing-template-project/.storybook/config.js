import { configure } from "@storybook/react";
import { addDecorator } from "@storybook/react";
import { withKnobs } from "@storybook/addon-knobs";
import { loadStories as loadLibraryStories } from '@seasketch/geoprocessing-client';

// automatically import all files ending in *.stories.tsx
const req = require.context("../src/client", true, /\.stories\..+$/);

function loadStories() {
  req.keys().forEach(req);
  loadLibraryStories();
}

configure(loadStories, module);
addDecorator(withKnobs);

