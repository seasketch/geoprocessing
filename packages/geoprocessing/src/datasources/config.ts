import { SupportedFormats } from "../types";

const vectorFormats: SupportedFormats[] = ["fgb", "json", "subdivided"];
const importSupportedVectorFormats: SupportedFormats[] = ["fgb", "json"];
const importDefaultVectorFormats: SupportedFormats[] = ["fgb"];

const rasterFormats: SupportedFormats[] = ["tif"];
const importSupportedRasterFormats: SupportedFormats[] = ["tif"];
const importDefaultRasterFormats: SupportedFormats[] = ["tif"];

const defaultDstPath = "data/dist";
const defaultBinPath = "data/bin";

/** Default datasource file location, relative to project root */
const defaultDatasourcesPath = "./project/datasources.json";

export const datasourceConfig = {
  vectorFormats,
  importSupportedVectorFormats,
  importDefaultVectorFormats,
  rasterFormats,
  importSupportedRasterFormats,
  importDefaultRasterFormats,
  defaultDstPath,
  defaultBinPath,
  defaultDatasourcesPath,
};
