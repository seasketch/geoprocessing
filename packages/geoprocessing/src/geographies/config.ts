const defaultDstPath = "data/dist";

/** Default file location, relative to project root */
const defaultSrcPath = "./project/geographies.json";

export const geographyConfig = {
  defaultSrcPath,
  defaultDstPath,
};
