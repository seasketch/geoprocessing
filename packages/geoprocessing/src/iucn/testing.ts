import { Sketch } from "../types/index.js";

export const genSketchWithActivities = (activities: string[]): Sketch => {
  return {
    type: "Feature",
    geometry: {
      type: "Polygon",
      coordinates: [],
    },
    properties: {
      id: "615bbbe9aac8c8285d50db2d",
      name: "mpa-test",
      updatedAt: "2021-10-05T02:43:53.326Z",
      createdAt: "2021-10-05T02:43:53.326Z",
      sketchClassId: "615b59e1aac8c8285d50d9b8",
      isCollection: false,
      userAttributes: [
        {
          label: "Allowed Activities",
          fieldType: "ChoiceField",
          exportId: "ACTIVITIES",
          value: activities,
        },
      ],
    },
  };
};
