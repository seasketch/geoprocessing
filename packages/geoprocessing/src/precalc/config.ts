const defaultDstPath = "data/dist";

/** Default file location, relative to project root */
const defaultSrcPath = "./data/dist/precalc.json";

export const precalcConfig = {
  defaultSrcPath,
  defaultDstPath,
};
