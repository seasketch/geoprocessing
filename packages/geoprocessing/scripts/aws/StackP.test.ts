import { App } from "aws-cdk-lib";
import path from "path";
import "@aws-cdk/assert/jest";
import { GeoprocessingStack, getHandlerPointer } from "./GeoprocessingStack";
import config from "./config";
import createTestProject from "../testing/createTestProject";
import { setupBuildDirs, cleanupBuildDirs } from "../testing/lifecycle";

const rootPath = `${__dirname}/__test__`;

describe("GeoprocessingStack - preprocessor only", () => {
  afterAll(() => cleanupBuildDirs(rootPath));

  it.only("should create a valid stack", async () => {
    const projectName = "preprocessor-only";
    const projectPath = path.join(rootPath, projectName);
    await setupBuildDirs(projectPath);

    const manifest = await createTestProject(projectName, ["preprocessor"]);

    expect(manifest.clients.length).toBe(0);
    expect(manifest.preprocessingFunctions.length).toBe(1);
    expect(manifest.geoprocessingFunctions.length).toBe(0);

    const app = new App();
    const stack = new GeoprocessingStack(app, projectName, {
      env: { region: manifest.region },
      projectName,
      manifest,
      projectPath,
    });

    // Check counts
    expect(stack).toCountResources("AWS::CloudFront::Distribution", 0);
    expect(stack).toCountResources("AWS::S3::Bucket", 2);
    expect(stack).toCountResources("AWS::ApiGateway::RestApi", 1);
    expect(stack).toCountResources("AWS::ApiGateway::Stage", 1);
    expect(stack).toCountResources("AWS::DynamoDB::Table", 2);
    expect(stack).toCountResources("AWS::Lambda::Function", 3);

    expect(stack).toHaveResourceLike("AWS::ApiGateway::Stage", {
      StageName: config.STAGE_NAME,
    });

    // Check shared resources
    expect(stack).toHaveResourceLike("AWS::ApiGateway::RestApi", {
      Name: `gp-${projectName}`,
    });
    expect(stack).toHaveResourceLike("AWS::S3::Bucket", {
      BucketName: `gp-${projectName}-results`,
    });
    expect(stack).toHaveResourceLike("AWS::S3::Bucket", {
      BucketName: `gp-${projectName}-datasets`,
    });
    expect(stack).toHaveResourceLike("AWS::Lambda::Function", {
      Handler: "serviceHandlers.projectMetadata",
      Runtime: config.NODE_RUNTIME.name,
    });
    expect(stack).toHaveResourceLike("AWS::DynamoDB::Table", {
      TableName: `gp-${projectName}-tasks`,
    });
    expect(stack).toHaveResourceLike("AWS::DynamoDB::Table", {
      TableName: `gp-${projectName}-estimates`,
    });

    // Check preprocessor resources
    expect(stack).toHaveResourceLike("AWS::Lambda::Function", {
      FunctionName: `gp-${projectName}-sync-${manifest.preprocessingFunctions[0].title}`,
      Handler: getHandlerPointer(manifest.preprocessingFunctions[0]),
      Runtime: config.NODE_RUNTIME.name,
    });
  });
});
