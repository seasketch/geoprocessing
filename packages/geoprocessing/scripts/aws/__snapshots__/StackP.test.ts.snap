// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoprocessingStack - preprocessor only should create a valid stack 1`] = `
Object {
  "Outputs": Object {
    "GpRestApiEndpointBF901973": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "GpRestApi3C403820",
            },
            ".execute-api.us-west-1.",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "GpRestApiDeploymentStageprodFDC6A6D1",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0ArtifactHash4A934180": Object {
      "Description": "Artifact hash for asset \\"1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0\\"",
      "Type": "String",
    },
    "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0S3Bucket5EA66AEF": Object {
      "Description": "S3 bucket for asset \\"1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0\\"",
      "Type": "String",
    },
    "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0S3VersionKeyD171B821": Object {
      "Description": "S3 key for asset version \\"1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "GpEstimatesTableBB595885": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "service",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": Array [
          Object {
            "AttributeName": "service",
            "KeyType": "HASH",
          },
        ],
        "TableName": "gp-preprocessor-only-estimates",
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
    "GpMetadataHandler5D0745A7": Object {
      "DependsOn": Array [
        "GpMetadataHandlerServiceRoleDefaultPolicyD81D593B",
        "GpMetadataHandlerServiceRole7107682D",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0S3Bucket5EA66AEF",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0S3VersionKeyD171B821",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0S3VersionKeyD171B821",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "FunctionName": "gp-preprocessor-only-metadata",
        "Handler": "serviceHandlers.projectMetadata",
        "Role": Object {
          "Fn::GetAtt": Array [
            "GpMetadataHandlerServiceRole7107682D",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "GpMetadataHandlerServiceRole7107682D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "GpMetadataHandlerServiceRoleDefaultPolicyD81D593B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "dynamodb:BatchGetItem",
                "dynamodb:GetRecords",
                "dynamodb:GetShardIterator",
                "dynamodb:Query",
                "dynamodb:GetItem",
                "dynamodb:Scan",
                "dynamodb:ConditionCheckItem",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "GpTasksTable2F8F158E",
                    "Arn",
                  ],
                },
                Object {
                  "Ref": "AWS::NoValue",
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GpMetadataHandlerServiceRoleDefaultPolicyD81D593B",
        "Roles": Array [
          Object {
            "Ref": "GpMetadataHandlerServiceRole7107682D",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GpPublicBucket989E1AE8": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": "gp-preprocessor-only-public",
        "CorsConfiguration": Object {
          "CorsRules": Array [
            Object {
              "AllowedHeaders": Array [
                "*",
              ],
              "AllowedMethods": Array [
                "HEAD",
                "GET",
              ],
              "AllowedOrigins": Array [
                "*",
              ],
              "Id": "cors-rule",
              "MaxAge": 3600,
            },
          ],
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "GpPublicBucketPolicyAAA4C049": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "GpPublicBucket989E1AE8",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Principal": "*",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Fn::GetAtt": Array [
                        "GpPublicBucket989E1AE8",
                        "Arn",
                      ],
                    },
                    "/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
    "GpRestApi3C403820": Object {
      "Properties": Object {
        "Description": "Serves API requests for preprocessor-only.",
        "Name": "gp-preprocessor-only",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "GpRestApiAccountFCFC4532": Object {
      "DependsOn": Array [
        "GpRestApi3C403820",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "GpRestApiCloudWatchRole263212C0",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "GpRestApiCloudWatchRole263212C0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "GpRestApiDeployment8BC1361A943afee03ab55d222c9364f9f3ba901d": Object {
      "DependsOn": Array [
        "GpRestApiGET037A17F9",
        "GpRestApiOPTIONS0D97A1CF",
        "GpRestApitestPreprocessorOPTIONSF3E96077",
        "GpRestApitestPreprocessorPOST8DD6678F",
        "GpRestApitestPreprocessor47A725A2",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "GpRestApi3C403820",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "GpRestApiDeploymentStageprodFDC6A6D1": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "GpRestApiDeployment8BC1361A943afee03ab55d222c9364f9f3ba901d",
        },
        "MethodSettings": Array [
          Object {
            "HttpMethod": "*",
            "ResourcePath": "/*",
            "ThrottlingBurstLimit": 20,
            "ThrottlingRateLimit": 40,
          },
        ],
        "RestApiId": Object {
          "Ref": "GpRestApi3C403820",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "GpRestApiGET037A17F9": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "GET",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "RequestTemplates": Object {
            "application/json": "{ \\"statusCode\\": \\"200\\" }",
          },
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:us-west-1:lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "GpMetadataHandler5D0745A7",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "GpRestApi3C403820",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "GpRestApi3C403820",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GpRestApiGETApiPermissionTestpreprocessoronlyGpRestApiBC18A662GETB34551A4": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "GpMetadataHandler5D0745A7",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-west-1:",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "GpRestApi3C403820",
              },
              "/test-invoke-stage/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GpRestApiGETApiPermissionpreprocessoronlyGpRestApiBC18A662GET7D17E368": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "GpMetadataHandler5D0745A7",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-west-1:",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "GpRestApi3C403820",
              },
              "/",
              Object {
                "Ref": "GpRestApiDeploymentStageprodFDC6A6D1",
              },
              "/GET/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GpRestApiOPTIONS0D97A1CF": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "OPTIONS",
        "Integration": Object {
          "IntegrationResponses": Array [
            Object {
              "ResponseParameters": Object {
                "method.response.header.Access-Control-Allow-Headers": "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent'",
                "method.response.header.Access-Control-Allow-Methods": "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'",
                "method.response.header.Access-Control-Allow-Origin": "'*'",
              },
              "StatusCode": "204",
            },
          ],
          "RequestTemplates": Object {
            "application/json": "{ statusCode: 200 }",
          },
          "Type": "MOCK",
        },
        "MethodResponses": Array [
          Object {
            "ResponseParameters": Object {
              "method.response.header.Access-Control-Allow-Headers": true,
              "method.response.header.Access-Control-Allow-Methods": true,
              "method.response.header.Access-Control-Allow-Origin": true,
            },
            "StatusCode": "204",
          },
        ],
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "GpRestApi3C403820",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "GpRestApi3C403820",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GpRestApitestPreprocessor47A725A2": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "GpRestApi3C403820",
            "RootResourceId",
          ],
        },
        "PathPart": "testPreprocessor",
        "RestApiId": Object {
          "Ref": "GpRestApi3C403820",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "GpRestApitestPreprocessorOPTIONSF3E96077": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "OPTIONS",
        "Integration": Object {
          "IntegrationResponses": Array [
            Object {
              "ResponseParameters": Object {
                "method.response.header.Access-Control-Allow-Headers": "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent'",
                "method.response.header.Access-Control-Allow-Methods": "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'",
                "method.response.header.Access-Control-Allow-Origin": "'*'",
              },
              "StatusCode": "204",
            },
          ],
          "RequestTemplates": Object {
            "application/json": "{ statusCode: 200 }",
          },
          "Type": "MOCK",
        },
        "MethodResponses": Array [
          Object {
            "ResponseParameters": Object {
              "method.response.header.Access-Control-Allow-Headers": true,
              "method.response.header.Access-Control-Allow-Methods": true,
              "method.response.header.Access-Control-Allow-Origin": true,
            },
            "StatusCode": "204",
          },
        ],
        "ResourceId": Object {
          "Ref": "GpRestApitestPreprocessor47A725A2",
        },
        "RestApiId": Object {
          "Ref": "GpRestApi3C403820",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GpRestApitestPreprocessorPOST8DD6678F": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "POST",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "RequestTemplates": Object {
            "application/json": "{ \\"statusCode\\": \\"200\\" }",
          },
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:us-west-1:lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "GpSyncHandlertestPreprocessorEC6E6881",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "GpRestApitestPreprocessor47A725A2",
        },
        "RestApiId": Object {
          "Ref": "GpRestApi3C403820",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "GpRestApitestPreprocessorPOSTApiPermissionTestpreprocessoronlyGpRestApiBC18A662POSTtestPreprocessorE21372D4": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "GpSyncHandlertestPreprocessorEC6E6881",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-west-1:",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "GpRestApi3C403820",
              },
              "/test-invoke-stage/POST/testPreprocessor",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GpRestApitestPreprocessorPOSTApiPermissionpreprocessoronlyGpRestApiBC18A662POSTtestPreprocessor8F7B916D": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "GpSyncHandlertestPreprocessorEC6E6881",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:us-west-1:",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "GpRestApi3C403820",
              },
              "/",
              Object {
                "Ref": "GpRestApiDeploymentStageprodFDC6A6D1",
              },
              "/POST/testPreprocessor",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "GpSyncHandlertestPreprocessorEC6E6881": Object {
      "DependsOn": Array [
        "GpSyncHandlertestPreprocessorServiceRole0D8ABBC3",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0S3Bucket5EA66AEF",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0S3VersionKeyD171B821",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0S3VersionKeyD171B821",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Description": "",
        "Environment": Object {
          "Variables": Object {
            "ESTIMATES_TABLE": Object {
              "Ref": "GpEstimatesTableBB595885",
            },
            "TASKS_TABLE": Object {
              "Ref": "GpTasksTable2F8F158E",
            },
            "publicBucketUrl": Object {
              "Fn::Join": Array [
                "",
                Array [
                  "https://s3.us-west-1.",
                  Object {
                    "Ref": "AWS::URLSuffix",
                  },
                  "/",
                  Object {
                    "Ref": "GpPublicBucket989E1AE8",
                  },
                ],
              ],
            },
          },
        },
        "FunctionName": "gp-preprocessor-only-sync-testPreprocessor",
        "Handler": "testPreprocessor.ts.handler",
        "MemorySize": 256,
        "Role": Object {
          "Fn::GetAtt": Array [
            "GpSyncHandlertestPreprocessorServiceRole0D8ABBC3",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 1,
      },
      "Type": "AWS::Lambda::Function",
    },
    "GpSyncHandlertestPreprocessorServiceRole0D8ABBC3": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "GpTasksTable2F8F158E": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AttributeDefinitions": Array [
          Object {
            "AttributeName": "id",
            "AttributeType": "S",
          },
          Object {
            "AttributeName": "service",
            "AttributeType": "S",
          },
        ],
        "BillingMode": "PAY_PER_REQUEST",
        "KeySchema": Array [
          Object {
            "AttributeName": "id",
            "KeyType": "HASH",
          },
          Object {
            "AttributeName": "service",
            "KeyType": "RANGE",
          },
        ],
        "TableName": "gp-preprocessor-only-tasks",
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
